src/ipc/ipc-common.c

```diff
@@ -190,8 +190,9 @@ void ipc_msg_send(struct ipc_msg *msg, void *data, bool high_priority)
 
 	key = k_spin_lock(&ipc->lock);
 
-	/* copy mailbox data to message */
-	if (msg->tx_size > 0 && msg->tx_size < SOF_IPC_MSG_MAX_SIZE) {
+	/* copy mailbox data to message if not already copied */
+	if ((msg->tx_size > 0 && msg->tx_size < SOF_IPC_MSG_MAX_SIZE) &&
+	    msg->tx_data != data) {
 		ret = memcpy_s(msg->tx_data, msg->tx_size, data, msg->tx_size);
 		assert(!ret);
 	}
```

Comment:
Please put this change in another patch as it is a bugfix or an enhancement which is not really related directly to the current patch.